tests/bitcoin_v2.rs

```diff
@@ -37,6 +38,36 @@ fn bitcoin_v2_submarine() {
     let invoice = "lntb510u1pnpuwgxpp5arh2aw92wv5vxmndm0704zu2qhj6k8xhg38dlylt2atcc3vcs88qdqgdverzvm2xqyjw5qcqp2sp52vgwm4s6pc2q38hxrma9h4ycgtn4kzaq0we0d4mjtq030fu6zsfqrzjq2gyp9za7vc7vd8m59fvu63pu00u4pak35n4upuv4mhyw5l586dvkf6vkyqq20gqqqqqqqqpqqqqqzsqqc9qyyssq6frgq8gly29thefxg83y57hv564npas5dqc0cslml2q3cj0usrdptqa5p2el3df0c0xc6raxnty06m745l87v4qausuq9xh0wj2x57cpxsxxcz".to_string();
     let refund_address = "tb1qq20a7gqewc0un9mxxlqyqwn7ut7zjrj9y3d0mu".to_string();
 
+    let boltz_api_v2 = BoltzApiClientV2::new(BOLTZ_TESTNET_URL_V2);
+
+    // Check for magic routing hint in invoice.
+    if let Some(route_hint) = find_magic_routing_hint(&invoice).unwrap() {
+        let mrh_resp = boltz_api_v2.get_mrh_bip21(&invoice).unwrap();
+
+        let bip21 = mrh_resp
+            .bip21
+            .parse::<Uri<'_, _>>()
+            .unwrap()
+            .require_network(bitcoin::Network::Testnet)
+            .unwrap();
+        let address_hash = sha256::Hash::hash(&Vec::from_hex(&bip21.address.to_string()).unwrap());
+        let msg = Message::from_digest_slice(address_hash.as_byte_array()).unwrap();
+
+        let receiver_sig = bitcoin::secp256k1::schnorr::Signature::from_slice(
+            &Vec::from_hex(&mrh_resp.signature).unwrap(),
+        )
+        .unwrap();
+
+        let receiver_pubkey = PublicKey::from_str(&route_hint.src_node_id.to_string())
+            .unwrap()
+            .inner;
+
+        secp.verify_schnorr(&receiver_sig, &msg, &receiver_pubkey.x_only_public_key().0)
+            .unwrap();
```

Comment:
This shouldn't just be in a test but a public function that is available for consumers of the library to 
call
We might want to integrate this in the library function that creates a swap

Reply:
There is no function as "create swap". Depends on the structure of the lib. Currently swap creating API is the test itself. Moved the MRH parts into it's own module `magic_routing.rs` and all the relevant functions are added there.